### DISCUSSION TITLE
Local development environment

### DISCUSSION THREAD

**Discussion by atanasenko** [omerfaruk-aran/esphome_samsung_hvac_bus#100] [General]
How do you setup your local development env? 

Previously, I've used vscode-server running as a container with esphome components mounted as a volume. Same volume is mounted to the esphome container for it to use. With this setup I only get syntax highlighting in ide, all the compilation problems are coming from the actual esphome build of the firmware.

So I've decided to move on to a proper desktop vscode running with a linux env (I'm using windows with wsl2 for that) so I can see the problems as I type.

If I'm using quickfixes suggested by vscode, I end up with a modified cpp_properties.json, with a few compilation errors here and there, and they get fixed with `"intelliSenseMode": "linux-gcc-x64"`, which is not right.

So what's the proper way to get vscode to understand what dep libs and platforms I'm dealing with? Platformio as with esphome?

**Comment by lanwin** [comment]
Currently I use the VSCode SSH integration to connect from my Windows machine to the Raspberry where my ESPhome instance lives. As root I use the path of my main ESPHome component. Then you can use git and also c++ and run sh commands. This is how I run the tests. 

Before that I installed msys g++ with runs the tests on windows. But that means to use two repositories and I had some case where I missed to push and pull while switching.

**Reply by atanasenko** [reply]
Running from console is ok. I mean how can I make vscode to do the compilation checks?

**Reply by lanwin** [reply]
I simply installed the Python and C++ extensions to the SSH remote side. Thats all. I the repo you find also the .vscode folder with the setting the C++ extension uses.

**Reply by atanasenko** [reply]
Does it somehow detect esphome includes automatically? They are not defined in the .vscode config.

**Reply by lanwin** [reply]
Yes, for the most part it does
